packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1417,12 +1298,39 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
       centerLayout(icon!, x);
     }
 
+    final double subtextBaseline = layout.subtextSize.ascent + layout.containerHeight;
+    final RenderBox? counter = this.counter;
+    final double helperErrorBaseline = helperError.getDistanceToBaseline(TextBaseline.alphabetic)!;
+    final double counterBaseline = counter?.getDistanceToBaseline(TextBaseline.alphabetic)! ?? 0.0;
+
+    double start, end;
```

Comment:
Could these be `late final`?

Reply:
they are updated after laying out each child.